Nitaicandra/CENPROJECT
# Db schemas + user route refactor

## User story
Set up MongoDB

## Summary of changes
This PR builds upon [Javi's PR ](https://github.com/Nitaicandra/CENPROJECT/pull/5)

- Created proper schemas for models and collection on Mongo
- Refactored the users route to work with the new schemas
- Decoupled authentication into a helper function
- Added a .dockerignore file to fix issues with the bcrypt lib and the docker container for server

## How to test
This change requires freshly built containers.
1. delete everything from Docker desktop (containers, images, and volumes). Rebuild the project
2. using postman, try to create a new business and customer accounts
3. using postman (or browser), try the GET endpoints and see if you can see your data
4. try to to create accounts with missing information and see if you get an error and the account (including auth info) have not been created

## some templates to use with postman
customer
{
        "username": "",
        "password": "",
        "firstName": "",
        "lastName": ",
        "address": "",
        "zipCode": "",
        "city": "",
        "state": "",
        "email": "",
        "phoneNumber": ""
   }
   
   business (with availability example)
  {
    "username": "",
    "password": "",
    "businessName": "",
    "address": "",
    "zipCode": "",
    "city": "",
    "state": "",
    "email": "",
    "phoneNumber": "",
    "availability": [{
        "Monday": [{"start": "9:00", "end": "12:00"}],
        "Tuesday": [{"start": "9:00", "end": "12:00"}, {"start": "13:00", "end": "17:00"}]
    }]
}
